server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/MailboxSetMethod.scala

```diff
@@ -110,7 +110,7 @@ class MailboxSetMethod @Inject()(serializer: Serializer,
           val (unparsableCreateRequests, createRequests) = parseCreateRequests(mailboxSetRequest)
           for {
             creationResults <- createMailboxes(mailboxSession, createRequests)
-            deletionResults <- deleteMailboxes(mailboxSession, mailboxSetRequest.destroy.getOrElse(Seq()))
+            deletionResults <- deleteMailboxes(mailboxSession, mailboxSetRequest.destroy.getOrElse(Seq()), mailboxSetRequest.onDestroyRemoveEmails.getOrElse(RemoveEmailsOnDestroy(false)))
```

Comment:
Let's modify deleteMailboxes to accept a MailboxSetRequest in order to prevent argument count growth and logic leaks....

That is a refactoring I did in one of my branches ;-)